App/Models/Payment.php

```diff
@@ -38,6 +39,14 @@ public static function statusToString($theStatus) {
     public static function isBeingAnalyzed($thePaymentInfo) {
         return $thePaymentInfo == null || ($thePaymentInfo['status'] != 3 && $thePaymentInfo['status'] != 4);
     }
+
+    public function getTotalPaid() {
+        if (isset(SELF::$total_paid)) return SELF::$total_paid;
```

Comment:
só por curiosidade, esse é o esquema de `cache` que tu comentou?

Reply:
Aham, é bem bobo mas ajuda, depois da pesquisa com o banco de dados ele armazeno o dado em uma variável, e nas próximas vezes usa o valor da variável. Não muda muito se for usado poucas vezes, mas dentro de loops com mais iterações pode fazer uma boa diferença :D
Acabei de perceber agora que faltou um parâmetro para ignorar o cache e forçar uma nova pesquisa, quando voltar a mexer nesse model eu implemento.